k256/src/arithmetic/projective.rs

```diff
@@ -250,6 +253,70 @@ impl From<AffinePoint> for ProjectivePoint {
     }
 }
 
+impl Normalize for ProjectivePoint {
+    fn batch_normalize_array<const N: usize>(points: &[Self; N]) -> [Self::AffineRepr; N] {
+        let mut zs = [FieldElement::ONE; N];
+
+        for i in 0..N {
+            if points[i].z != FieldElement::ZERO {
+                // Even a single zero value will fail inversion for the entire batch.
+                // Put a dummy value (above `FieldElement::ONE`) so inversion succeeds
+                // and treat that case specially later-on.
+                zs[i] = points[i].z;
+            }
+        }
+
+        // This is safe to unwrap since we assured that all elements are non-zero
+        let zs_inverses = <FieldElement as Invert>::batch_invert_array(&zs).unwrap();
+
+        let mut affine_points = [AffinePoint::IDENTITY; N];
+        for i in 0..N {
+            if points[i].z != FieldElement::ZERO {
+                // If the `z` coordinate is non-zero, we can use it to invert;
+                // otherwise it defaults to the `IDENTITY` value in initialization.
+                affine_points[i] = points[i].to_affine_internal(zs_inverses[i])
+            }
+        }
+
+        affine_points
+    }
+
+    #[cfg(not(feature = "alloc"))]
```

Comment:
@tarcieri I'm not sure how to get rid of this.
And if I change this to return `vec`, it will not compile obviously as we're in `(not(feature = "alloc")`